rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -483,6 +496,33 @@ private String cleanDisplayedErrorMessage(String errorMessage) {
         return (errorMessage);
     }
 
+    /**
+     * Start the nodes acquisition of the node source
+     *
+     * @param sessionId a valid session id
+     * @param nodeSourceName the name of the node source to start
+     * @return the result of the action, possibly containing the error message
+     * @throws NotConnectedException
+     */
+    @Override
+    @PUT
+    @Path("nodesource/deploy")
+    @Produces("application/json")
+    public NSState deployNodeSource(@HeaderParam("sessionid") String sessionId,
+            @FormParam("nodeSourceName") String nodeSourceName) throws NotConnectedException {
+        ResourceManager rm = checkAccess(sessionId);
+        NSState nsState = new NSState();
+        try {
+            nsState.setResult(rm.deployNodeSource(nodeSourceName).getBooleanValue());
+        } catch (RuntimeException ex) {
+            nsState.setResult(false);
+            nsState.setErrorMessage(cleanDisplayedErrorMessage(ex.getMessage()));
+            nsState.setStackTrace(StringEscapeUtils.escapeJson(getStackTrace(ex)));
+        } finally {
+            return nsState;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this return statement from this finally block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1143)